### PR TITLE
tracing: Bump otel version and update tracing exporter

### LINKED ISSUE #226: Tracing: Update to latest OpenTelemetry version
We are currently on `0.18.0` but in the meantime there has already been a couple of stable releases, latest one being [`1.3.0`](https://pkg.go.dev/go.opentelemetry.io/otel). Since the package organization, the public API etc. has change between version, this will require some manual 'reshuffling'.

### PR DESCRIPTION
**Pull request by PhilipGough**
Bumps otel libs from v0.18.0 -> v1.5.0 and adds corresponding changes to tracing init function.

Fixes #226

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,12 +34,12 @@ require (
...
 	github.com/prometheus/common v0.32.1
 	github.com/prometheus/prometheus v1.8.2-0.20210621150501-ff58416a0b02
-	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.18.0
-	go.opentelemetry.io/contrib/propagators v0.18.0
-	go.opentelemetry.io/otel v0.18.0
-	go.opentelemetry.io/otel/exporters/trace/jaeger v0.18.0
-	go.opentelemetry.io/otel/sdk v0.18.0
-	go.opentelemetry.io/otel/trace v0.18.0
+	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.30.0
+	go.opentelemetry.io/contrib/propagators v0.22.0
```
Comment (matej-g) [inline go.mod:38]: The versioning is quiet confusing :thinking: I don't understand why `otelhttp` doesn't have stable version yet.

But I think we can update propagators to
```suggestion
	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp v0.30.0
	go.opentelemetry.io/contrib/propagators/jaeger v1.5.0
```

**Review by matej-g — COMMENTED** [review]
This is looking good to me, thanks @PhilipGough! I have small suggestions.

File: main.go [hunk 2]
```diff
@@ -223,7 +223,7 @@ func main() {
 	logger := logger.NewLogger(cfg.logLevel, cfg.logFormat, cfg.debug.name)
 	defer level.Info(logger).Log("msg", "exiting")
 
-	tp, closer, err := tracing.InitTracer(
+	tp, err := tracing.InitTracer(
```
[the remark this answers is not in the document]
Comment (PhilipGough) [reply main.go:226]: Yeah, agree, was thinking the same when I was making the change to be honest but was just trying to keep changes related to bumping the version. Happy to do the refactor here though.

**Review by matej-g — APPROVED** [review]
Looking great @PhilipGough! Thank you taking care of this!

**Pull request merged**